File: scripts/nitro/template/android/src/main/java/com/margelo/nitro/modulename/ModuleName.kt
package com.margelo.nitro.{{cxxNamespace}}

import com.facebook.proguard.annotations.DoNotStrip
import com.margelo.nitro.core.Promise

@DoNotStrip
class {{modulePascalCase}} : Hybrid{{modulePascalCase}}Spec() {
  override fun hello(params: {{modulePascalCase}}Params): Promise<{{modulePascalCase}}Result> {
    val result = {{modulePascalCase}}Result(success = true, data = "Hello, ${params.message}!")
    return Promise.resolved(result)
  }
}

File: scripts/nitro/template/ios/ModuleName.swift
import NitroModules

class {{modulePascalCase}}: Hybrid{{modulePascalCase}}Spec {
    
    public func hello(params: {{modulePascalCase}}Params) throws -> Promise<{{modulePascalCase}}Result> {
        let result = {{modulePascalCase}}Result(success: true, data: "Hello, \(params.message)!")
        return Promise.resolved(withResult: result)
    }
}
